From 740f6e038785c00e9449506a0e43d14f2767a0dc Mon Sep 17 00:00:00 2001
From: Phillip Schichtel <pschichtel@users.noreply.github.com>
Date: Sun, 28 Feb 2016 16:08:43 +0100
Subject: [PATCH 1/2] Update init.pp

- replaced repos-parameter with origins
- added an ensure parameter that's passed through to the package
- remove ensure from the service, the subscription is enough
---
 manifests/init.pp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index a5dcb89..f1b8270 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -1,6 +1,7 @@
 class unattended_upgrades(
+  $ensure                       = 'installed'
   $period                       = 1,                                             # Update period (in days)
-  $repos                        = {},                                            # Repos to upgrade
+  $origin                       = [],                                            # Origin patterns
   $blacklist                    = [],                                            # Packages to not update
   $email                        = '',                                            # Email for update status
   $autofix                      = true,                                          # Ensure updates keep getting installed
@@ -17,12 +18,12 @@
   $apt_path = '/etc/apt/apt.conf.d/20auto-upgrades'
   $package = 'unattended-upgrades'
 
-  if $::operatingsystem !~ /^(Debian|Ubuntu)$/ {
-    fail("${::operatingsystem} is not supported.")
+  if $::osfamily != 'Debian' {
+    fail("${::osfamily} is not supported.")
   }
 
   package { $package:
-    ensure => latest,
+    ensure => $ensure,
   }
 
   file { $conf_path:
@@ -42,7 +43,6 @@
   }
 
   service { $package:
-    ensure    => running,
     subscribe => [ File[$conf_path], File[$apt_path], Package[$package], ],
   }
 }

From 3e503c09982e0014ee518b6c344734ebfdbd0374 Mon Sep 17 00:00:00 2001
From: Phillip Schichtel <pschichtel@users.noreply.github.com>
Date: Sun, 28 Feb 2016 16:10:04 +0100
Subject: [PATCH 2/2] Render the new origins array instead of constructing them

---
 templates/unattended-upgrades.erb | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/templates/unattended-upgrades.erb b/templates/unattended-upgrades.erb
index a77b22a..cfb2a33 100644
--- a/templates/unattended-upgrades.erb
+++ b/templates/unattended-upgrades.erb
@@ -1,8 +1,8 @@
 // Automatically upgrade packages from these origin patterns
 Unattended-Upgrade::Origins-Pattern {
-<% scope['unattended_upgrades::repos'].each do |repo, opts| -%>
-        "origin=${distro_id},archive=<%= repo %><% if (defined? opts) %>,label=<%= opts['label'] %><% end %>";
-<% end %>
+<%- scope['unattended_upgrades::origins'].each do |origin| -%>
+        "<%= origin %>";
+<%- end -%>
 }; 
 
 // List of packages to not update
